endpoint.go

```diff
@@ -70,7 +64,7 @@ func NewEndpoint(serviceName string, hostPort string) (*Endpoint, error) {
 
 // NewEndpointOrNil tries to create a new endpoint and returns it. On error
 // nil will be returned.
-func NewEndpointOrNil(serviceName string, hostPort string) *Endpoint {
+func NewEndpointOrNil(serviceName string, hostPort string) *model.Endpoint {
```

Comment:
I am curious about this, couldn't we return an Endpoint pointer and an error instead of a nil?

Reply:
This is a convenience wrapper around NewEndpoint which just what you say... this allows to plug in NewEndpointOrNil straight into the tracer / span options and just be nil if failing to create.

Reply:
Not sure whether this is a valid use case. I believe when you are creating an `Endpoint` there is a firm intention to do it (meaning that you know it is a web service, not a cli service)

Reply:
In the end it comes down to: do you regard the creation of an endpoint to be best effort (it can fail on host name lookup as an example) or critical enough to need to write code to handle error cases in case it does fail.

Reply:
Not sure here. In my mind `serviceName` and `hostPort` are things that could be checked statically.

Reply:
given that such a wrapper can be simple constructed by consumers I will remove this one

Reply:
👍